src/main/kotlin/no/nav/familie/ef/sak/infotrygd/PeriodeService.kt

```diff
@@ -159,36 +159,34 @@ class PeriodeService(
             aktivitet = andelsVedtak.aktivitet,
             periodeType = andelsVedtak.periodeType,
             barn = behandlingsbarn.map { BehandlingsbarnMedOppfyltAleneomsorg(personIdent = it.personIdent, fødselTermindato = it.fødselTermindato) },
-            harAktivitetsplikt =
-                harAktivitetsplikt(
-                    aktivitet = andelsVedtak.aktivitet,
-                    periodetype = andelsVedtak.periodeType,
-                ),
+            harAktivitetsplikt = andelsVedtak.aktivitet.harAktivitetsplikt(),
         )
     }
 
-    private fun harAktivitetsplikt(
-        aktivitet: AktivitetType,
-        periodetype: VedtaksperiodeType,
-    ): Boolean {
-        if (periodetype === VedtaksperiodeType.HOVEDPERIODE || periodetype === VedtaksperiodeType.NY_PERIODE_FOR_NYTT_BARN) {
-            return aktivitet === AktivitetType.FORSØRGER_I_ARBEID ||
-                aktivitet === AktivitetType.FORSØRGER_REELL_ARBEIDSSØKER ||
-                aktivitet === AktivitetType.FORSØRGER_I_UTDANNING ||
-                aktivitet === AktivitetType.FORSØRGER_ETABLERER_VIRKSOMHET
-        }
-
-        if (periodetype === VedtaksperiodeType.UTVIDELSE) {
-            return aktivitet === AktivitetType.UTVIDELSE_FORSØRGER_I_UTDANNING
+    private fun AktivitetType.harAktivitetsplikt(): Boolean =
+        when (this) {
+            AktivitetType.FORSØRGER_I_ARBEID -> true
```

Comment:
Foreslår å skrive om til

AktivitetType.FORSØRGER_I_ARBEID, AktivitetType.FORSØRGER_I_UTDANNING, ... , -> true
og tilsvarende for false